src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -140,6 +140,67 @@ public async Task<DataTable> ExecuteQuery(string query, string stampName, string
             var cluster = await GetClusterNameFromStamp(stampName);
             await AddQueryInformationToMetadata(query:query, cluster:cluster, operationName:operationName);
             return await _kustoClient.ExecuteQueryAsync(Helpers.MakeQueryCloudAgnostic(_kustoMap, query), _kustoMap.MapCluster(cluster) ?? cluster, _kustoMap.MapDatabase(_configuration.DBName) ?? _configuration.DBName, requestId, operationName, _queryStartTime, _queryEndTime);
+        }
+
+        private Dictionary<string, Tuple<string, string>> clusterStampNameMapping = new Dictionary<string, Tuple<string, string>>() 
+        {
+            {"wawswus", new Tuple<string, string>("waws-prod-bay-153", "WestUS") },
+            {"wawseus", new Tuple<string, string>("waws-prod-blu-189", "EastUS") },
+            {"wawscus", new Tuple<string, string>("waws-prod-dm1-187", "CentralUS") },
+            {"wawsweu", new Tuple<string, string>("waws-prod-am2-329", "WestEurope") },
+            {"wawsneu", new Tuple<string, string>("waws-prod-db3-169", "NorthEurope") },
+            {"wawseas", new Tuple<string, string>("waws-prod-hk1-029", "EastAsia") }

```

Comment:
are these stamps specific to Static Web Apps only?

Reply:
Please add a comment explaining what these stamps are.

Reply:
Nope, these are just random stamp names so that ExecuteQuery is able to pickup the correct cluster based on the failover logic. One random stamp name per region.

Reply:
So this is a superset over the `ExecuteQuery` method except detector author doensn't need to provide `StampName` since SWA don't have concepts of stamps?

Reply:
That is correct @hforeste. SWA had a requirement where they wanted to query BlueRidgeDiagnostics table which does not exist in wawsneu, wawsweu, and wawseas clusters. The data however can be in any of the other regions. The answer is to union the clusters where the table is known to exist. Union however fails because it is a cross cluster query. This method is a counterpart of ExecuteQueryOnAllAppServiceClusters but like the name suggests, you can target which clusters the query should run on and the queries are fired directly at those clusters eliminating the need for union via cross cluster queries.